plutus-tx/src/PlutusTx/Foldable.hs

```diff
@@ -139,12 +139,14 @@ product = getProduct #. foldMap Product
 
 -- | Plutus Tx version of 'Data.Foldable.foldrM'.
 foldrM :: (Foldable t, Haskell.Monad m) => (a -> b -> m b) -> b -> t a -> m b
+-- See Note [Haskell.Monad]
 foldrM f z0 xs = foldl c Haskell.return xs z0
   where c k x z = f x z Haskell.>>= k
         {-# INLINE c #-}
 
 -- | Plutus Tx version of 'Data.Foldable.foldlM'.
 foldlM :: (Foldable t, Haskell.Monad m) => (b -> a -> m b) -> b -> t a -> m b
+-- See Note [Haskell.Monad]
 foldlM f z0 xs = foldr c Haskell.return xs z0
   where c x k z = f z x Haskell.>>= k
         {-# INLINE c #-}
```

Comment:
Ditto. Ironically though, this function is less inappropriate in Plutus than it is in Haskell (lazy left folds are stupid).